internal/component/prometheus/remote/queue/network/manager.go

```diff
@@ -0,0 +1,171 @@
+package network
+
+import (
+	"context"
+
+	"github.com/grafana/alloy/internal/runtime/logging/level"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/alloy/internal/component/prometheus/remote/queue/types"
+	"github.com/vladopajic/go-actor/actor"
+)
+
+// manager manages loops. Mostly it exists to control their lifecycle and send work to them.
+type manager struct {
+	connectionCount uint64
+	loops           []*loop
+	metadata        *loop
+	logger          log.Logger
+	inbox           actor.Mailbox[*types.TimeSeriesBinary]
+	metaInbox       actor.Mailbox[*types.TimeSeriesBinary]
+	configInbox     actor.Mailbox[types.ConnectionConfig]
+	self            actor.Actor
+	cfg             types.ConnectionConfig
+	stats           func(types.NetworkStats)
+	metaStats       func(types.NetworkStats)
+}
+
+var _ types.NetworkClient = (*manager)(nil)
+
+var _ actor.Worker = (*manager)(nil)
+
+func New(cc types.ConnectionConfig, logger log.Logger, seriesStats, metadataStats func(types.NetworkStats)) (types.NetworkClient, error) {
+	s := &manager{
+		connectionCount: cc.Connections,
+		loops:           make([]*loop, 0, cc.Connections),
+		logger:          logger,
+		// This provides blocking to only handle one at a time, so that if a queue blocks
+		// it will stop the filequeue from feeding more.
+		inbox:       actor.NewMailbox[*types.TimeSeriesBinary](actor.OptCapacity(1)),
+		metaInbox:   actor.NewMailbox[*types.TimeSeriesBinary](actor.OptCapacity(1)),
+		configInbox: actor.NewMailbox[types.ConnectionConfig](),
+		stats:       seriesStats,
+		cfg:         cc,
+	}
+
+	// start kicks off a number of concurrent connections.
+	var i uint64
```

Comment:
why is i initialized outside of the for loop?

Reply:
Its because connection count is a uint64, I will switch to `i := uint64(0)` since that is clearer.